src/events/ready.ts

```diff
@@ -1,19 +1,38 @@
 import { Events } from 'discord.js'
 
+import { Environment } from '../config.js'
 import { EventHandlerConfig } from '../types/EventHandlerConfig.js'
 
 export default {
   eventName: Events.ClientReady,
   once: true,
   execute: async (client) => {
-    console.log(`Helloworld, Online as ${client.user?.tag}.`)
+    console.log(`[ready] Now online as ${client.user?.tag}.`)
     const commands_data = [...client.commands.values()].map(
       (command) => command.data,
     )
+
+    // Set guild commands
+    try {
+      const guild = client.guilds.cache.get(Environment.GUILD_ID)
+      if (!guild) {
+        throw new Error(`Guild ${Environment.GUILD_ID} not found`)
+      }
+      await guild.commands.set(commands_data)
+      console.info(`[ready] Guild commands registered on ${guild.name}`)
+    } catch (error) {
+      console.error('[ready] Unable to set guild commands:', error)
+    }
+
+    // Clear global commands
     try {
-      await client.application?.commands.set(commands_data)
+      const commands = await client.application?.commands.fetch()
+      for (const command of commands?.values() || []) {
+        await command.delete()
+        console.info(`[ready] Deleted global command ${command.name}`)
+      }
     } catch (error) {
-      console.log(error)
+      console.error('[ready] Unable to clear application commands:', error)
     }
```

Comment:
We only need to clear global command once right? Can we run that externally? After we run that we should no longer need this code.

Reply:
@Leomotors 

> We only need to clear global command once right?

yes, it will run just once

> Can we run that externally? After we run that we should no longer need this code.

yes, we can. but some other developers who ran the older version of the code will have global commands lying around and they will be confused.

it is easier to just have the bot clear the global application commands automatically, rather than communicating with other devs that they need to run an external command, bet that they will actually read the message, and be prepared to answer the question.

factoring in human elements, let's let the robot do the boring task.